### PR TITLE
[EXTENSION] Rework

### PR DESCRIPTION
**Pull request by horumyy**
alright amikes, this pr brings a lot of shit, lets break it down:

- UI 
  - Steps state are now more visually appealing
  - Added Tabs for future CRUD (`NewFlow.tsx`and `LoadFlow.tsx`)
  - Added remove and edit buttons for `StepsCard`
  - Removed add button
  - Upload button changed position, content and style
  - Bigger icons on step's state
  - Changed Loading lottie
  - Added lil sandwich
  - Added Inter font all around the app
  

_white background is not present on build_
Preview

Screenshot 2024-05-28 at 9 29 43 p m

- Contexts
  - Added `StateContext` in order to keep track of global state variables
  - Added `StepsContext` in order to keep all the steps globally, including its methods as `handleAddStep`, `handleRemoveStep` and `handleEditStep`

- Other
  - Moved most of the `App.tsx` content to its own component
  - Added `Tabs.tsx` component from shadcn/ui
  - Created `StepsCard` component

### CODE REVIEW

File: chrome-extension/src/screens/NewFlow.tsx [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+      }
+    };
+
+    window.addEventListener("keydown", handleKeyPress);
+    return () => window.removeEventListener("keydown", handleKeyPress);
+  }, [newStep]);
+
+  React.useEffect(() => {
+    getStatusFromBG(handleExtensionResponse);
+  }, []);
```
Comment (Int-Angel) [inline chrome-extension/src/screens/NewFlow.tsx:36]: I think this should be inside the StepsContext, what do you think?

File: chrome-extension/src/contexts/StepsContext.tsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+      currentView: "",
+      state: "completed",
+    },
+    {
+      index: 3,
+      prompt: "Failed",
+      currentView: "",
+      state: "failed",
+    },
+  ];
```
Comment (Int-Angel) [inline chrome-extension/src/contexts/StepsContext.tsx:44]: Just remove before merge

File: chrome-extension/src/screens/NewFlow.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    setNewStep(e.target.value);
+  };
+
+  const handleExtensionResponse = (response: ApplicationStateType) => {
+    if (response === null) return;
+    setSteps(response.steps);
+    setState(response.state);
+
+    console.log("Response from background: ", response);
+  };
```
Comment (Int-Angel) [inline chrome-extension/src/screens/NewFlow.tsx:74]: I also think this should be part of the context

**Review by Int-Angel — APPROVED** [review]
Amazing job on this rework! The UI looks amazing and I like the new context

File: chrome-extension/src/App.tsx [hunk 4]
```diff
@@ -37,178 +21,40 @@ function App() {
...
-        if (newStep.trim() === "") return;
-        handleAddItem();
-      }
-    };
-
-    window.addEventListener("keydown", handleKeyPress);
-    return () => window.removeEventListener("keydown", handleKeyPress);
-  }, [newStep]);
-
-  const handleExtensionResponse = (response: ApplicationStateType) => {
```
Comment (CarlosCRG19) [inline chrome-extension/src/App.tsx:57]: getting an error cause this function is not defined but still used in line 15

File: chrome-extension/src/App.tsx [hunk 5]
```diff
@@ -1,27 +1,11 @@
...
-import { Button } from "./components/Button";
-import {
-  IoReorderFourOutline,
-  IoCheckmarkCircleOutline,
-  IoCloseCircleOutline,
-} from "react-icons/io5";
-import { StepsType } from "./types/steps";
-import { sendStepsToBG, getStatusFromBG } from "./utils/ExecuteActionsUtils";
-import { ApplicationStateType, ApplicationStates } from "./types/application";
-import { RequestType } from "./types/request";
```
Comment (CarlosCRG19) [inline chrome-extension/src/App.tsx:14]: don't remove this import

File: chrome-extension/src/App.tsx [hunk 6]
```diff
@@ -1,27 +1,11 @@
...
-  const [steps, setSteps] = React.useState<StepsType[]>([]);
-  const [newStep, setNewStep] = React.useState("");
-  const [state, setState] = React.useState<ApplicationStates>("idle");
-
-  //save reference for dragItem and dragOverItem
-  const dragItem = React.useRef<number | null>(null);
-  const dragOverItem = React.useRef<number | null>(null);
+  const [selectedTab, setSelectedTab] = React.useState("newFlow");
 
   React.useEffect(() => {
```
Comment (CarlosCRG19) [inline chrome-extension/src/App.tsx:10]: although i wonder if this effect is still needed

File: chrome-extension/src/App.tsx [hunk 7]
```diff
@@ -37,178 +21,40 @@ function App() {
...
-          }}
-          disabled={state === "running"}
-        >
-          {state === "running" ? <LoadingLottie /> : "Send"}
-        </Button>
-      </div>
+            onClick={() => setSelectedTab("loadFlow")}
+            value="loadFlow"
+          >
+            Load flow
```
Comment (CarlosCRG19) [inline chrome-extension/src/App.tsx:48]: this should be `saved flows`

File: chrome-extension/src/App.tsx (hunk 6 above)
Comment (Int-Angel) [reply chrome-extension/src/App.tsx:10]: yesss, but I think should be inside the app context

File: chrome-extension/src/screens/NewFlow.tsx (hunk 3 above)
Comment (horumyy) [reply chrome-extension/src/screens/NewFlow.tsx:74]: im not sure about this, shouldnt the context be only for functions related to that state?

File: chrome-extension/src/App.tsx (hunk 5 above)
Comment (horumyy) [reply chrome-extension/src/App.tsx:14]: i ended up moving the useEffect to another file so i removed it

File: chrome-extension/src/App.tsx (hunk 6 above)
Comment (horumyy) [reply chrome-extension/src/App.tsx]: i ended up placing it inside the`NewFlow.tsx` file

File: chrome-extension/src/App.tsx (hunk 7 above)
Comment (horumyy) [reply chrome-extension/src/App.tsx]: done

File: chrome-extension/src/App.tsx (hunk 4 above)
Comment (horumyy) [reply chrome-extension/src/App.tsx:57]: done

File: chrome-extension/src/contexts/StepsContext.tsx (hunk 2 above)
Comment (horumyy) [reply chrome-extension/src/contexts/StepsContext.tsx]: removed

File: chrome-extension/src/screens/NewFlow.tsx (hunk 1 above)
Comment (horumyy) [reply chrome-extension/src/screens/NewFlow.tsx]: mmm, i think the contexts should stay with only the functions related to that state, but im not sure about this, im new to contexts

**Review by CarlosCRG19 — APPROVED** [review]

**Pull request merged**